tensorflow/python/kernel_tests/conv2d_transpose_test.py

```diff
@@ -24,8 +24,12 @@
 from tensorflow.python.client import device_lib
```

Comment:
The C++ shape fn can do this correctly, but has a bug - it is explicitly checking input_tensor instead of just calling MakeShapeFromShapeTensor  (MakeShapeFromShapeTensor used to not be able to handle this case correctly, but now does).

Can you instead change core/ops/nn_ops.cc -> the InputTensorShapeOrUnknown function doesn't need to call input_tensor and check for nullptr, it can just call MakeShapeFromShapeTensor.

Reply:
Thanks @cwhipkey! The `MakeShapeFromShapeTensor` works great. I didn't notice it before. The PR has been updated.

Reply:
I think all the callers to InputTensorShapeOrUnknown would benefit from this change - would you mind changing that function instead, to not do the nullptr check and to have the WithRank addition?

Reply:
Thanks @cwhipkey, done.